### PR TITLE
Add a console reporter

### LINKED ISSUE #36: Include a Telemetry.Metrics.ConsoleReporter
This could be a good debug tool that basically prints the events as they come. For example, in my app I know there is "vm.memory", but I don't know the keys, metadata and so on. The goal of the console reporter is to show what is being consumed and what isn't.

It can also work as a good example for other reporters.

Thoughts?

### PR DESCRIPTION
**Pull request by josevalim**
Closes #36.

### CODE REVIEW

File: lib/telemetry_metrics/console_reporter.ex [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      Event name: vm.memory
+      All measurements: %{binary: 100, total: 200}
+      All metadata: %{}
+
+      Metric measurement: :binary (last_value)
+      With value: 100 byte
+      And tag values: %{}
+
+      Metric measurement: :total (counter)
+      With value: 200
```
Comment (arkgil) [inline lib/telemetry_metrics/console_reporter.ex:33]: I'm wondering if we shouldn't always print `1` as measurement value for counters. Or maybe add an info that counter's measurement is not relevant.

**Comment by arkgil** [comment]
@josevalim Looks good to me! I've left one comment for discussion :)

File: lib/telemetry_metrics/console_reporter.ex [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  end
+
+  defp info(int) when is_number(int), do: ""
+  defp info(_), do: " (WARNING! measurement should be a number)"
+
+  defp unit(:unit), do: ""
+  defp unit(unit), do: " #{unit}"
+
+  defp metric(Telemetry.Metrics.Counter), do: "counter"
+  defp metric(Telemetry.Metrics.Distribution), do: "counter"
```
Comment (josevalim) [inline lib/telemetry_metrics/console_reporter.ex]:
```suggestion
  defp metric(Telemetry.Metrics.Distribution), do: "distribution"
```

File: lib/telemetry_metrics/console_reporter.ex (hunk 1 above)
Comment (josevalim) [reply lib/telemetry_metrics/console_reporter.ex:33]: Even better: I won't show the value for them. Good catch!

**Review by arkgil — APPROVED** [review]

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**